### ISSUE TITLE
🐛 Cannot use biome surpessions inside templating language comments

### ISSUE THREAD

**Issue by TeaDrinkingProgrammer** [biomejs/biome#9805]
### Environment information

I am using Biome for linting [Tera](https://keats.github.io/tera/docs/) templates for a [Zola](https://www.getzola.org/) blog. The syntax is largely the same as Jinja/Django templates. I am trying to surpess an error with the comment syntax `{# comment #}` , so that the surpression is not shown in the final HTML, but this crashes Biome.

<details>

```bash

$ biome check --write --unsafe
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_9
Message: Tried to cast node with kind HTML_BOGUS as `"biome_html_syntax::generated::nodes::HtmlRoot"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

templates/_base.html internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind HTML_BOGUS as `"biome_html_syntax::generated::nodes::HtmlRoot"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.

```

</details>

### What happened?

1. Make a Biome config with
```json
	"html": {
		"parser": {
			"interpolation": true
		}
	}
```

2. Try to run `biome check --write --unsafe` with the following snippet

```html
{# biome-ignore-start lint/a11y/useValidLang: tags #}
<html lang="{% block lang %}{% endblock lang %}">
{# biome-ignore-end lint/a11y/useValidLang: tags #}
```

3. Error

### Expected result

No error is thrown and the error is ignored.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
